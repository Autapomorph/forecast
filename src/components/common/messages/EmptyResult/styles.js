import styled from 'styled-components';

import Message from '../Message';

// eslint-disable-next-line import/prefer-default-export
export const StyledEmptyListMessage = styled(Message)``;
