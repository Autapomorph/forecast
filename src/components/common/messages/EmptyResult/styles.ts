import styled from 'styled-components/macro';

import Message from 'components/common/messages/Base';

export const EmptyResultMessage = styled(Message)`
  margin: 20px 0;
`;
