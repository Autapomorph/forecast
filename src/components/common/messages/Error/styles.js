import styled from 'styled-components';

import Message from '../Message';

// eslint-disable-next-line import/prefer-default-export
export const StyledErrorMessage = styled(Message)`
  background: var(--bg-error);
  color: var(--error-text-color);
  font-size: 1.3rem;
`;
