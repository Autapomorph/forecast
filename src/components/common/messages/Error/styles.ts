import styled from 'styled-components/macro';

import Message from 'components/common/messages/Message';

export const StyledErrorMessage = styled(Message)`
  background: var(--bg-error);
  color: var(--error-text-color);
`;
