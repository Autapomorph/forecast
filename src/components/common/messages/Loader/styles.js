import styled from 'styled-components';

import Message from '../Message';

// eslint-disable-next-line import/prefer-default-export
export const StyledLoaderMessage = styled(Message)`
  margin: 10px 0;
`;
