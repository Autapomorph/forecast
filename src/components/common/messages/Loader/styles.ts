import styled from 'styled-components/macro';

import Message from 'components/common/messages/Base';

export const LoaderMessage = styled(Message)`
  margin: 10px 0;
  font-size: 1.5rem;
`;
