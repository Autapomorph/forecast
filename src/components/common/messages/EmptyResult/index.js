import React from 'react';

import { StyledEmptyResultMessage } from './styles';

const EmptyResult = ({ children, ...props }) => (
  <StyledEmptyResultMessage {...props}>{children}</StyledEmptyResultMessage>
);

export default EmptyResult;
