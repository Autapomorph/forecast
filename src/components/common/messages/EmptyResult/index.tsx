import React from 'react';

import * as S from './styles';

const EmptyResult: React.FC = ({ children, ...props }): React.ReactElement => (
  <S.EmptyResultMessage {...props}>{children}</S.EmptyResultMessage>
);

export default EmptyResult;
