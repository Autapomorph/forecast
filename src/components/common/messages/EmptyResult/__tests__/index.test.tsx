import React from 'react';
import { shallow } from 'enzyme';

import EmptyResult from '..';

import * as S from '../styles';

describe('EmptyResult tests', () => {
  it('renders properly', () => {
    const wrapper = shallow(<EmptyResult>Test</EmptyResult>);

    expect(wrapper).toMatchSnapshot();
  });

  it('should render children', () => {
    const text = 'text';
    const wrapper = shallow(<EmptyResult>{text}</EmptyResult>);

    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find(S.EmptyResultMessage).text()).toEqual(text);
  });
});
