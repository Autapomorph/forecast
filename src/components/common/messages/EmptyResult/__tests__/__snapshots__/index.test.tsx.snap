// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EmptyResult tests renders properly 1`] = `
<styles__EmptyResultMessage>
  Test
</styles__EmptyResultMessage>
`;

exports[`EmptyResult tests should render children 1`] = `
<styles__EmptyResultMessage>
  text
</styles__EmptyResultMessage>
`;
